docs/contributing/adding_an_extension.md

```diff
@@ -0,0 +1,44 @@
+# Adding an extension
+
+Psalm tries to support most extensions with stubs so that projects requiring extensions can be analyzed without having
+those extensions installed at runtime. When an extension isn't already supported by Psalm, if it is loaded at runtime
+Psalm will use reflection to support it as best as possible, but it's always better to have stubs. In addition to
+allowing analysis without having the extension installed, stubs can be more accurate. The gmp and decimal extensions
+have a lot of functions that use `numeric-string` instead of `string`, and many extension like Ds, DOM, and others use
+templates. Most extension stubs can be improved by using Psalm features that aren't possible from reflection alone.
+
+## Using the stub generator
+
+We try to make adding support for an extension as easy as possible by providing a simple way to generate stubs with
+reflection, but keep in mind that reflection can miss a lot of useful information:
+
+ - Which `@throws` tags should be added to a method
+ - Psalm specific types like `numeric-string`, `int<0, max>`, or `non-empty-list`.
+ - [Templates](../annotating_code/templated_annotations.md)
+
+To use the extension stub generator, download the latest phar at
+https://github.com/AndrolGenhald/php-extension-stub-generator/releases and run `php-extension-stub-generator.phar dump
```

Comment:
Does someone want to create psalm/php-extension-stub-generator and move this there?

Reply:
We'd have to ask Matt I think. Let's do that when it's finished

Reply:
Well, I think it is finished, more or less :stuck_out_tongue:
I was thinking this would be merged as mostly a documentation/process update.